code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/AEPMessage.java

```diff
@@ -257,6 +263,16 @@ public void show(final boolean withMessagingDelegateControl) {
     /** Dismisses the message. */
     @Override
     public void dismiss() {
+        if (messagesMonitor.dismiss()) {
```

Comment:
i believe this should be if `!messageMonitor.dismiss()` - this method returns `true` if a message was being shown, and `false` if no dismissal was necessary